pkg/generate/app/cmd/newapp_test.go

```diff
@@ -1265,6 +1265,11 @@ func TestEnsureValidUniqueName(t *testing.T) {
 			expected: []string{"one", "one-1", "two"},
 		},
 		{
+			name:     "non-standart characters",
+			input:    []string{"Emby.One", "test-_test", "_-_"},
+			expected: []string{"embyone", "test-test", ""},
```

Comment:
not sure what's valid, I'm just suggesting having a test case that will validate your "strip the dash if the regex leaves the name prefixed with a dash" logic.